java/server/src/org/openqa/selenium/remote/server/commandhandler/UploadFile.java

```diff
@@ -60,8 +64,20 @@ public void execute(HttpRequest req, HttpResponse resp) throws IOException {
           "Expected there to be only 1 file. There were: " +
           (allFiles == null ? 0 : allFiles.length)));
     } else {
+      // IE requires multiple uploads to all be from the same directory - move files into one location
+      Path uploads_dir = Files.createDirectories(tempDir.toPath().getParent().resolve("remote_uploads"));
+      Path dest = uploads_dir.resolve(allFiles[0].getName());
+
+      if (Files.exists(dest)){
+        // IE maintains a lock on previously uploaded files so we can't delete/overwrite - move it instead
+        Path prevFileDir = session.getFileSystem().createTempDir("prevupload", "file").toPath();
+        Files.move(dest, prevFileDir.resolve(allFiles[0].getName()));
```

Comment:
Of course, you're totally screwed if someone tries to upload a file with the same name _again_. Why not just pick a random file name?

Reply:
We can't pick a random file name because the file name is sent to the browser. There are tests that confirm it's the same as the name of the file selected by the user.

There isn't an issue of someone trying to upload the same file twice in different uploads because of the previous `move`, and if they try to upload the same file name twice in the same multiple upload then those would have to have been the same file since IE requires the files to be from the same directory so no issue.  On other browsers if 2 files with the same name from different directories were selected then there would be an issue.  If that's a reasonable concern then this code could be only used when the browser is IE (if we actually have access to that info at this point in the code).